config/initializers/6_preload_stis.rb

```diff
@@ -0,0 +1,7 @@
+Rails.application.config.to_prepare do
+  # Preload all classes which use Foreman::STI and using registration methods
+  # E.g. Base.register_type(BMC)
+  # Some constants that use such classes may be defined before all the related classes/models are loaded and registered
+  # E.g. InterfaceTypeMapper::ALLOWED_TYPE_NAMES
+  Rails.autoloaders.main.eager_load_dir("#{Rails.root}/app/models/nic")
```

Comment:
`Rails.root` is a `Pathname`, so `Rails.root.join('app', 'models', 'nic')` looks cleaner to me.